import-scripts/preconsume_problematic_samples.sh

```diff
@@ -168,20 +169,20 @@ function post_slack_message() {
 date
 check_args
 make_tmp_dir_if_necessary
+set_cvr_fetch_url_prefix
 failed_to_consume_problematic_events_sample_list=() # temporary code
 succeeded_to_consume_problematic_events_sample_list=() # temporary code
 failed_to_consume_problematic_metadata_sample_list=() # temporary code
 succeeded_to_consume_problematic_metadata_sample_list=() # temporary code
-set_cvr_fetch_url_prefix
-consume_hardcoded_samples # temporary code
-fetch_currently_queued_samples
-detect_samples_with_problematic_events
-detect_samples_with_problematic_metadata
-exit_if_no_problems_detected
-failed_to_consume_problematic_events_sample_list=()
-succeeded_to_consume_problematic_events_sample_list=()
-failed_to_consume_problematic_metadata_sample_list=()
-succeeded_to_consume_problematic_metadata_sample_list=()
-attempt_to_consume_problematic_samples
+while :
+do
+    consume_hardcoded_samples # temporary code
+    fetch_currently_queued_samples
+    detect_samples_with_problematic_events
+    detect_samples_with_problematic_metadata
+    exit_if_no_problems_detected
```

Comment:
This function currently does a sys.exit() system call, which would terminate the program immediately (before sending reports to slack or logging the problematic samples which were consumed.

I'd suggest turning this into a predicate function .. perhaps called "problems_were_detected" which returned 0 if there were problems and non-zero (1) if no problems were detected. Then this line could become:
    if ! problems_detected; then
        break
    fi

So the loop would end but we would continue to the log_actions and post_slack_message. However --- we probably also want to skip posting a slack message if there were absolutely no problematic samples consumed at all. So we may need a function to detect how many problematic samples actually were consumed (regardless of the number of times the loop was iterated).

Also .. we might need to check that at the end of attempting to consume problematic samples we blank out the files containing the sample ids which need to be consumed --- resetting them for the next loop iteration. I'm not sure if this is necessary, but it may be.

Reply:
good catch, I'll work on implementing that
to your last point, the files containing the sample ids to be consumed are rewritten on each loop iteration (currently the `consume_hardcoded_samples` function does this)